pkg/util/log/logger.go

```diff
@@ -0,0 +1,344 @@
+package log
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/sirupsen/logrus"
+)
+
+// FieldLogger Wraps the StdLogger, and provides logrus methods for logging with fields
+type FieldLogger interface {
+	StdLogger
+	WithField(key string, value interface{}) FieldLogger
+	WithFields(fields logrus.Fields) FieldLogger
+	WithError(err error) FieldLogger
+}
+
+// StdLogger interface for logging methods found in the go standard library logger.
+type StdLogger interface {
+	logrus.StdLogger
+
+	Debug(args ...interface{})
+	Debugf(format string, args ...interface{})
+	Debugln(args ...interface{})
+
+	Error(args ...interface{})
+	Errorf(format string, args ...interface{})
+	Errorln(args ...interface{})
+
+	Info(args ...interface{})
+	Infof(format string, args ...interface{})
+	Infoln(args ...interface{})
+
+	Warn(args ...interface{})
+	Warnf(format string, args ...interface{})
+	Warnln(args ...interface{})
+}
+
+// Redactor interface for redacting log messages
+type Redactor interface {
```

Comment:
Should the `Redactor` be added to the `FieldLogger` interface?

Reply:
The AMPGW team added these, would want to check to see how they are using this.